src/main/scala/firrtl/passes/ZeroWidth.scala

```diff
@@ -2,16 +2,63 @@
 
 package firrtl.passes
 
-import scala.collection.mutable
 import firrtl.PrimOps._
 import firrtl.ir._
 import firrtl._
 import firrtl.Mappers._
 
+/**
+  * Transform that replaces zero width mems before running the ZeroWidth transform.
+  * Dealing with mems is a bit tricky for that pass because the address, en, clk ports
+  * of the memory are not width zero even if data is.
+  *
+  * This pass replaces memories with a DefWire() bundle that contains the address, en,
+  * clk, and data fields implemented as zero width wires. Running the ZeroWidth
+  * transform afterwards will remove these dangling references properly.
+  *
+  * This pass is private because it should only be run by the ZeroWidth transform
+  */
+private object ZeroWidthMemRemove extends Transform {
```

Comment:
How do you feel if this could just be part of the ZeroWidth transform as a separate onModule function, not as its own private transform? In general we discourage transforms to be objects instead of classes as well.